packages/typescript-private/device-server/src/device-host/device-host.resign-app-file.ts

```diff
@@ -0,0 +1,89 @@
+import { ResignShPath } from '../res-map';
+
+@Injectable()
+export class DeviceHostResignAppFileService {
+  private queueLock = new AsyncLock();
+  private readonly client = axios.create();
+
+  constructor(private readonly logger: DoguLogger) {
+    setAxiosErrorFilterToIntercepter(this.client);
+  }
+
+  public async queueResign(message: ResignAppFileRequestBody): Promise<ResignAppFileResponseBodyData> {
+    const result = await this.queueLock.acquire(
+      message.filePath,
+      async () => {
+        return await this.resign(message);
+      },
+      {
+        timeout: Milisecond.t15Minutes,
+        maxOccupationTime: Milisecond.t15Minutes,
+      },
+    );
+    if (result.result !== 'success') {
+      this.logger.error('DeviceHostResignAppFileService.queueResign error', { result });
+    }
+    return result;
+  }
+
+  private async resign(message: ResignAppFileRequestBody): Promise<ResignAppFileResponseBodyData> {
+    const appPath = message.filePath;
+    const doguHome = HostPaths.doguHomePath;
+    const configsPath = HostPaths.configsPath(doguHome);
+    const identityName = env.APPLE_RESIGN_IDENTITY_NAME;
+    const provisioningProfilePath = HostPaths.resignProvisoningProfilePath(configsPath);
+
+    if (process.platform !== 'darwin') {
+      return { result: 'not-macos' };
+    }
+
+    if (!appPath.endsWith('.ipa')) {
+      return { result: 'not-ipa' };
+    }
+
+    if (0 == identityName.length) {
+      return { result: 'no-identity-specified' };
+    }
+    if (!fs.existsSync(provisioningProfilePath)) {
+      return { result: 'no-provisioning' };
+    }
+
+    if (!fs.existsSync(HostPaths.doguTempPath())) {
+      await fs.promises.mkdir(HostPaths.doguTempPath(), { recursive: true });
+    }
+
+    const identityResult = await ChildProcess.execIgnoreError('security find-identity', {}, this.logger);
+    if (!identityResult.stderr.includes(identityName) && identityResult.stderr.includes(identityName)) {
+      return { result: 'no-identity-exists' };
+    }
+
+    const contents = await fs.promises.readFile(ResignShPath, { encoding: 'utf-8' });
+    const shellPath = path.resolve(HostPaths.doguTempPath(), `resign-${uuidv4()}.sh`);
+    await fs.promises.writeFile(shellPath, contents, { encoding: 'utf-8' });
+    await fs.promises.chmod(shellPath, 0o755);
+
+    const resignedAppPath = path.resolve(HostPaths.doguTempPath(), `resign-${uuidv4()}.ipa`);
+    const args = [shellPath, appPath, `"${identityName}"`, '-p', provisioningProfilePath, resignedAppPath].join(' ');
+
+    this.logger.info('DeviceHostResignAppFileService.resign', { args });
+    const result = await ChildProcess.exec(args, {}, this.logger);
+    this.logger.info('DeviceHostResignAppFileService.resign done', { result });
+    await fs.promises.rm(appPath, { recursive: true, force: true });
+    await fs.promises.cp(resignedAppPath, appPath, { recursive: true, force: true });
+
+    await fs.promises.rm(shellPath, { recursive: true, force: true });
```

Comment:
If you fail due to unexpected reasons during the re-signing process, there will be several random files.
What do you think about tying it in blocks of 
`try{ resign... } finally {if file exist then delete... }`
?